### ISSUE TITLE
Invoke-IcingaCheckMemory : has no access to read perf counters, but returns OK

### ISSUE THREAD

**Issue by K0nne** [Icinga/icinga-powershell-plugins#25]
Hello,

I'm about to test the powershell plugins via powershell remoting against a test system. My user has currently no access to read perf counters on the target host.

When I tested  **Invoke-IcingaCheckMemory** and **Invoke-IcingaCheckCPU**  and stumbled upon this problem I was surprised that both checks showed a different plugin result:

**Invoke-IcingaCheckMemory**   
returns OK but show internal errors
```
invoke-command -computername FQDN -ScriptBlock { Use-Icinga; Invoke-IcingaCheckMemory; } -credential $cred -authentication negotiate

Get-CimInstance: Access denied
RuntimeException: Attempted to divide by zero.
MethodInvocationException: Exception calling "Format" with "2" argument(s): "Value cannot be null.
Parameter name: args"
[OK] Check package "Memory Usage"
| 'memory_percent_used'=%;;;0;100 'used_bytes'=0B;;;0
0
```

**Invoke-IcingaCheckCPU**
returns UNKNOWN and an helpful advice about the problem
```
invoke-command -computername FQDN -ScriptBlock { Use-Icinga; Invoke-IcingaCheckCPU; } -credential $cred -authentication negotiate

[UNKNOWN]: Icinga Permission Error was thrown: PerformanceCounter

A Plugin failed to fetch Performance Counter information. This may be caused when the used Service User is not permitted to access these information. To fix this, please add the User the Icinga Agent is running on into the "Performance Log Users" group and restart the service.
```

I think Invoke-IcingaCheckMemory should behave like Invoke-IcingaCheckCPU and also point to the permission problem. Those hints are very helpful, btw 👍 

Thank you!

**Comment by LordHepipud** [comment]
Thank you very much for the feedback. I honestly never expected WMI/CIM instances to be used without a default permission set to be honest 😛

But you are right, this should be addressed and fixed. I will have a look on this on how to make this work properly.

Thank you a lot !

**Comment by K0nne** [comment]
My company is very security-aware. I have to go step by step in sense of access rights until it works 😅

During my journey so far I also found out, that the automatic unit detection in this check does not work, if there's no access to the CIM class Win32_LocalMemory. I think I'm the only one ever with this problem 😂  

But when I think about it, maybe it whould be good to return the not accessible class(es) in the plug-in output to give a hint where to search.

**Comment by LordHepipud** [comment]
Yeah, I fully agree on this. Just have to figure out a way to do this the most easiest way 😄

**Comment by LordHepipud** [comment]
I'm currently about to have a look on this issue and would require one additional input:
When you run

```powershell
try {
    Get-CimInstance -Class Win32_LocalMemory;
} catch {
$_.Exception.StatusCode;
$_.Exception.NativeErrorCode;
}
```

what do you get as output in PowerShell? With this I can add a proper error handling for this. In addition, what happens if you run the same command as WMI:

```powershell
try {
    Get-WmiObject -Class Win32_LocalMemory;
} catch {
$_.Exception.HResult;
}
```

Thank you in advance!
